### PR TITLE
feat: tabsbar component

### PR DESCRIPTION
**Pull request by devartstar**
fixes: #12 

Progress made -

- [x] Creating tabsbar component.
- [x] Replace DOM manupulation using string with Vue logic
- [x] Fixing the change in `circuitName` in Properties Panel on switching between tabs.
- [x] Sorting of tabs by dragging. (using Vue-SortableJS)
- [ ] CSS Refactoring.

### CODE REVIEW

**Comment by devartstar** [comment]
I have linked the `tabs-bar circuitName` with the `properties-panel circuitName`
@abstrekt @ayan-biswas0412 can you please review this PR

**Comment by abstrekt** [comment]
I'm able to delete all circuits from the properties panel, and then not able to add any any circuit.
image

**Comment by abstrekt** [comment]
can you also compare the nav and tabs bar with the production version? some CSS differences can be observed ..

**Comment by abstrekt** [comment]
When we select a tab from the tabs bar the tabs name field is focused and text selected in the properties tab, but this is not happening with the last tab.
image
Works fine with other tabs
image

**Comment by abstrekt** [comment]
Low hanger: When we select the create button , the input text resets to "Untitled-Circuit" before closing, this shouldn't be visible in the UI.
image

**Comment by abstrekt** [comment]
When we click + to create a new sub-circuit, lets focus and select the default text "Untitled-Circuit" just like in production, as most of the time user will try to change this, so it is a good UX. 
image

File: src/components/MessageBox/messageBox.vue [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+
+onUpdated(() => {
+    inputVal.value = 'Untitled-Circuit'
+})
+
+const props = defineProps({
+    messageText: { type: String, default: '' },
+    isPersistent: { type: Boolean, default: false },
+    buttonList: { type: Array, default: undefined },
+    inputList: { type: Array, default: undefined },
```
Comment (abstrekt) [inline src/components/MessageBox/messageBox.vue:50]: Check default values here, for type Array default should be empty array if required is set to false, any specific reasons for undefined? undefined should rarely be used as values in JS.

File: src/components/Panels/PropertiesPanel/ModuleProperty/ModuleProperty.vue [hunk 2]
```diff
@@ -30,7 +30,8 @@ import HelpButton from '#/components/Panels/Shared/HelpButton.vue'
 import ElementProperty from '#/components/Panels/PropertiesPanel/ModuleProperty/ElementProperty/ElementProperty.vue'
 import ProjectProperty from '#/components/Panels/PropertiesPanel/ModuleProperty/ProjectProperty/ProjectProperty.vue'
 import SubcircuitPropert from '#/components/Panels/PropertiesPanel/ModuleProperty/SubcircuitProperty/SubcircuitProperty.vue'
-import { toRefs } from '@vue/reactivity'
+import { ref, toRefs } from '@vue/reactivity'
+import { onMounted } from '@vue/runtime-core'
```
Comment (abstrekt) [inline src/components/Panels/PropertiesPanel/ModuleProperty/ModuleProperty.vue:34]: I guess we can directly import `from "vue"`

File: src/components/Panels/PropertiesPanel/ModuleProperty/ProjectProperty/ProjectProperty.vue [hunk 3]
```diff
@@ -80,6 +81,20 @@ import { getProjectName } from '#/simulator/src/data/save'
...
 import simulationArea from '#/simulator/src/simulationArea'
 import InputGroups from '#/components/Panels/Shared/InputGroups.vue'
+import { ref } from '@vue/reactivity'
+import { onMounted } from '@vue/runtime-core'
+const circuitId = ref(0)
+const circuitName = ref('Untitled-Cirucit')
+
+onMounted(() => {
+    // checking if circuit or tab is switched
+    setInterval(() => {
```
Comment (abstrekt) [inline src/components/Panels/PropertiesPanel/ModuleProperty/ProjectProperty/ProjectProperty.vue:91]: we'd need a better way to do this, setInterval won't be a good option here,

File: src/components/TabsBar/TabsBar.vue [hunk 4]
```diff
@@ -0,0 +1,216 @@
...
+                >
+                    <span class="circuitName noSelect">
+                        {{ truncateString(scopeList[element.id].name, 18) }}
+                    </span>
+                    <span
+                        id="scope.id"
+                        class="tabsCloseButton"
+                        @click="closeCircuit($event, element)"
+                    >
+                        x
```
Comment (abstrekt) [inline src/components/TabsBar/TabsBar.vue:35]: we'll need to use a proper icon here,

File: src/components/TabsBar/TabsBar.vue [hunk 5]
```diff
@@ -0,0 +1,216 @@
...
+                    </span>
+                </div>
+            </template>
+        </draggable>
+        <button @click="createNewCircuit()">&#43;</button>
+    </div>
+    <MessageBox
+        v-model="SimulatorState.dialogBox.create_circuit"
```
Comment (abstrekt) [inline src/components/TabsBar/TabsBar.vue:43]: I'm not able to find the MessageBox component accepting v-model, can you add some context here?

File: src/components/TabsBar/TabsBar.vue [hunk 6]
```diff
@@ -0,0 +1,216 @@
...
+    createNewCircuitScope,
+    deleteCurrentCircuit,
+    getDependenciesList,
+    scopeList,
+    switchCircuit,
+} from '#/simulator/src/circuit'
+import MessageBox from '#/components/MessageBox/messageBox.vue'
+import { useState } from '#/store/SimulatorStore/state'
+
+const SimulatorState = useState()
```
Comment (abstrekt) [inline src/components/TabsBar/TabsBar.vue:75]: I can notice there are some wrong patterns on how we are using store here, lets have a call regarding this?

File: src/components/MessageBox/messageBox.vue (hunk 1 above)
Comment (devartstar) [reply src/components/MessageBox/messageBox.vue:50]: Fixed !

File: src/components/TabsBar/TabsBar.vue (hunk 4 above)
Comment (devartstar) [reply src/components/TabsBar/TabsBar.vue:35]: Done, replaced it with a close icon.

File: src/components/Panels/PropertiesPanel/ModuleProperty/ProjectProperty/ProjectProperty.vue (hunk 3 above)
Comment (devartstar) [reply src/components/Panels/PropertiesPanel/ModuleProperty/ProjectProperty/ProjectProperty.vue:91]: will create an issue for it to be looked at afterward.

**Review by vedant-jain03 — APPROVED** [review]

**Review by abstrekt — APPROVED** [review]

**Pull request merged**